### PR TITLE
Add support for attributes on record fields and union cases

### PR DESCRIPTION
**Pull request by Smaug123**
This should be squashed in, due to a force-push I did in the history of `master`.

Of particular concern is https://github.com/nickcowle/TeqCrate/compare/fantomas2...attributes?expand=1#diff-e86e3744613b71183e675c83fe071c4532bebd7847fab1bd9f60d83baf33a259R267 where I *believe* I have exhaustively covered the available cases for a union, but am not certain.

### CODE REVIEW

File: TeqCrate.Test/TestUnion.fs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+        attributes.["Case3"]
+        |> List.exactlyOne
+        |> fun data -> data.AttributeType
+        |> shouldEqual typeof<Bar>
+
+        attributes.["Case4"]
+        |> List.map (fun data -> data.AttributeType)
+        |> List.sortBy (fun ty -> ty.Name)
+        |> shouldEqual [ typeof<Bar> ; typeof<Foo> ]
```
Comment (saul) [inline TeqCrate.Test/TestUnion.fs:76]: I would expect that the types defined in the source are the same in which they are defined in IL - why did you add the sort?

**Review by saul — APPROVED** [review]
Seems perfectly cromulent to me

File: TeqCrate.Test/TestUnion.fs (hunk 1 above)
Comment (Smaug123) [reply TeqCrate.Test/TestUnion.fs:76]: Just so that I didn't have to bother thinking about order, really. Pure laziness.

**Review by nickcowle — APPROVED** [review]
